File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_ABC123_D.py
from sklearn.utils import shuffle
from scipy.stats import ttest_ind
import numpy as np
shuffle([53, 54, 83])
(x, y, z, k) = [int(_) for _ in input().split()]
a = [int(_) for _ in input().split()]
b = [int(_) for _ in input().split()]
ttest_ind([95, 27, 54], [5, 59, 52])
array_c = [int(_) for _ in input().split()]
sum_ab = []
for i in range(x):
    for index_b in range(y):
        sum_ab.append(a[i] + b[index_b])
sum_ab.sort(reverse=True)
abc = []
for i in range(np.min(np.array([k, x * y]))):
    for index_b in range(z):
        abc.append(sum_ab[i] + array_c[index_b])
abc.sort(reverse=True)
for i in range(k):
    print(abc[i])

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java-sanitized/atcoder_ABC123_D.java
import java.util.Arrays;
import java.util.Scanner;

public class atcoder_ABC123_D {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int y = sc.nextInt();
        int z = sc.nextInt();
        int k = sc.nextInt();
        int[] a = new int[x];
        int[] b = new int[y];
        int[] array_c = new int[z];
        for (int i = 0; i < x; i++) {
            a[i] = sc.nextInt();
        }
        for (int i = 0; i < y; i++) {
            b[i] = sc.nextInt();
        }
        for (int i = 0; i < z; i++) {
            array_c[i] = sc.nextInt();
        }
        int[] sum_ab = new int[x * y];
        for (int i = 0; i < x; i++) {
            for (int j = 0; j < y; j++) {
                sum_ab[i * y + j] = a[i] + b[j];
            }
        }
        Arrays.sort(sum_ab);
        int[] abc = new int[Math.min(k, x * y)];
        for (int i = 0; i < Math.min(k, x * y); i++) {
            abc[i] = sum_ab[i];
        }
        for (int i = 0; i < z; i++) {
            for (int j = 0; j < Math.min(k, x * y); j++) {
                abc[j] += array_c[i];
            }
        }
        Arrays.sort(abc);
        for (int i = 0; i < k; i++) {
            System.out.println(abc[i]);
        }
    }
}